src/main.cpp

```diff
@@ -1 +1,149 @@
+        for (int z = 0; z < 256; ++z)
+        {
+            floorGrid.set({x, 0, z}, (x + z) % 2 == 0 ? black : white);
+        }
+    }
+    renderer->setPalette(palette);
+
+    auto floorRendererGrid = renderer->upload(floorGrid);
+    auto floor = assetManager.store<data::Grid>("floor", data::Usage::Static,
+                                                data::Grid{
+                                                    .grid = std::move(floorGrid),
+                                                    .rendererGrid = floorRendererGrid,
+                                                });
+
+    // Spawn the floor.
+    cmds.create(ecs::Grid{floor, {-128.0f, -1.0f, -128.0f}}, ecs::LocalToWorld{}, ecs::Position{}, ecs::Scale{4.0f});
+}
+
+void spawnCar(Commands cmds, data::AssetManager& assetManager) {
+    auto car = assetManager.load<data::Grid>("car");
+    cmds.create(Car{},
+                ecs::Grid{car, {-float(car->grid.getSize().x) / 2.0f, 0.0f, -float(car->grid.getSize().z) / 2.0f}},
+                ecs::LocalToWorld{}, ecs::Position{{56.0f, 0.0f, 56.0f}}, ecs::Rotation{});
+}
+
+void spawnCamera(Commands cmds, plugins::ActiveCamera& activeCamera) {
+    auto camera =
+        cmds.create(ecs::Camera{60.0f, 0.1f, 1000.0f}, ecs::LocalToWorld{}, ecs::Position{{0.0f, 40.0f, -70.0f}},
+                    ecs::Rotation{glm::quatLookAt(glm::vec3{0.0f, 0.0f, 1.0f}, glm::vec3{0.0f, 1.0f, 0.0f})});
+
+    activeCamera.entity = camera.entity();
+}
+
+/*
+void followCarSystem(core::ecs::Query<ecs::Camera&, ecs::Position&, ecs::Rotation&> cameraQuery, core::ecs::Query<Car&, ecs::Position&, ecs::Rotation&> carQuery, plugins::ActiveCamera& activeCamera, const DeltaTime& deltaTime) {
+    //query activeCamera?
+    //change rotation && position
+}
+*/
+
+static void turnOnLight(gl::Frame& frame)
+{
+    frame.ambient({0.1f, 0.1f, 0.1f});
+
+    // Add a directional light to the frame.
+    glm::quat directionalLightRotation = glm::quat(glm::vec3(glm::radians(45.0f), glm::radians(45.0f), 0));
+    frame.light(core::gl::DirectionalLight(directionalLightRotation, glm::vec3(1), 1.0f));
+}
+
+void moveSystem(core::ecs::Query<Car&, ecs::Position&, ecs::Rotation&> query, const DeltaTime& deltaTime)
+{
+    for (auto [entity, car, position, rotation] : query)
+    {
+        car.vel = rotation.quat * glm::vec3(2.0f, 0.0f, 2.0f);
+        car.angVel = 0.3f;
+
+        position.vec += deltaTime.value * car.vel;
+        rotation.quat = glm::angleAxis(deltaTime.value * car.angVel, glm::vec3(0.0f, 1.0f, 0.0f)) * rotation.quat;
+    }
+}
+
+int main(int argc, char** argv)
+{
+    auto cubos = Cubos(argc, argv);
+
+    cubos.addPlugin(plugins::windowPlugin);
+    cubos.addPlugin(plugins::rendererPlugin);
+    cubos.addPlugin(plugins::fileSettingsPlugin);
+
+    cubos.addResource<data::AssetManager>()
+        .addComponent<ecs::LocalToWorld>()
+        .addComponent<ecs::Position>()
+        .addComponent<ecs::Rotation>()
+        .addComponent<ecs::Scale>()
```

Comment:
Unnecessary since the `transformPlugin` does this automatically - `rendererPlugin` includes it.